### ISSUE TITLE
feat(emails): add the ability to track deliverability rates

### ISSUE THREAD

**Issue by DevinWalker** [WordImpress/Give#2841]
## Issue Overview
This WP.org topic has a question regarding "How do I know my emails were delivered from Give to the recipient": https://wordpress.org/support/topic/new-donation-email-notification-not-being-received/

It would be nice to have the ability to track whether an email has been delivered properly to the recipients inbox. It would also be nice if you could then track whether it has been opened and clicked, but that's likely an additional feature request.

## Expected Behavior
Admins should be able to see whether an email has been delivered or not.

## Current Behavior
Admins are unaware if an email is sent properly without using a service like SendGrid or Amazon SES.

## Possible Solution
Track responses from the mail servers to see if an email has been accepted or rejected.

**Comment by slewisma** [comment]
Need to decide how deep to go with this feature. Sent okay is different than delivered which is different than opened. Only the first can be reported without implementing a much more complete tracking system. Open tracking is only possible if the user (and their IT department or email provider in some cases) allows images or other outbound links in the message.

Unless you're prepared to implement an Email Service-like pixel tracking capability, all you can do is report back based on the return codes from sending. There are a lot of reasons an email might not be delivered that will still have an okay returned from the WP_Mail / PHP Mailer / SMTP send functions.

**Comment by mathetos** [comment]
I think the best we can do is add a "Email Sent" Note in the Donation Notes section of each individual donation, similarly to how we do that with MailChimp and Google Analytics. Additionally, we could add a Log of sorts that collates those Notes. 

But like @slewisma said, unless we want to consider a email saas service, out options within the plugin directly are limited.

**Comment by mariadanieldeepak** [comment]
@DevinWalker,

How about Cc'ing the site admin? Also, I see in WordPress forum that you have suggested Email Cop to log emails. I have a better recommendation - https://wordpress.org/plugins/email-log/

**Comment by mathetos** [comment]
Hi @mariadanieldeepak -- just a point of clarification. We don't recommend Email Cop for logging, it's for troubleshooting because it BLOCKS the emails from being sent and allows you to see them visually. We recommend Email Log often for logging, it's a great tool. I use it on every site I manage.

**Comment by mariadanieldeepak** [comment]
Thank you for the clarification @mathetos.

**Comment by slewisma** [comment]
I agree with the reason for closing this and further suggest that trackers can also be added in the email templates so that marketing automation tools (Mautic, Hubspot, Marketo, etc.) can be used to track the emails even if plain old SMTP is used to send them.
